### PR TITLE
Add more pprof functions to server and agent’s admin server

### PR DESCRIPTION
**Pull request by zqzten**
This PR added some missing pprof functions to the server and agent’s admin server.
It also updated .gitignore to include some common paraphernalia.

### CODE REVIEW

File: cmd/agent/app/server.go [hunk 1]
```diff
@@ -152,6 +152,10 @@ func (a *Agent) runAdminServer(o *options.GrpcProxyAgentOptions) error {
 	if o.EnableProfiling {
 		muxHandler.HandleFunc("/debug/pprof", util.RedirectTo("/debug/pprof/"))
 		muxHandler.HandleFunc("/debug/pprof/", pprof.Index)
+		muxHandler.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+		muxHandler.HandleFunc("/debug/pprof/profile", pprof.Profile)
```
Comment (jkh52) [inline cmd/agent/app/server.go:156]: Is this alone meaningful? (Given that the agent does not integrate with [cpu profiling](https://pkg.go.dev/runtime/pprof#StartCPUProfile))

Comment (zqzten) [reply cmd/agent/app/server.go:156]: Yeah, `pprof.Profile` does proactive cpu profiling so we don't have to explicitly integrate that elsewhere in the code.

File: cmd/agent/app/server.go [hunk 2]
```diff
@@ -152,6 +152,10 @@ func (a *Agent) runAdminServer(o *options.GrpcProxyAgentOptions) error {
 	if o.EnableProfiling {
 		muxHandler.HandleFunc("/debug/pprof", util.RedirectTo("/debug/pprof/"))
 		muxHandler.HandleFunc("/debug/pprof/", pprof.Index)
+		muxHandler.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
```
Comment (cheftako) [inline cmd/agent/app/server.go:155]: You can make an argument that all the pprof methods can expose sensitive information and that is why we require it to be explicitly enabled. That having been said cmdline both seems to potentially expose things the most (eg. AuthenticationAudience and anything we unwittingly add later) but also the least useful given that the cmdline flags are already being logged at startup. Additionally its much easier to do things like mask fields we consider to be too sensitive in the logs. Much harder to do that on pprof/cmdline. I'm ok with the rest but I would prefer we not expose cmdline.

Comment (zqzten) [reply cmd/agent/app/server.go]: That sounds reasonable, removed.

File: cmd/agent/app/server.go (hunk 1 above)
Comment (jkh52) [reply cmd/agent/app/server.go:155]: ack, thanks. I'll defer to @cheftako to approve.

**Pull request merged**